CYF-ITP-Sheffield|Kazeem Hamzat|Module-User-Focused-Data|Week2

HamzatKazeem opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

HamzatKazeem committed:
Initial Commit

- Create form
style input type elements

HamzatKazeem committed:
Add validation controls to form element

-Use regex for pattern checking on input type
-autofocus on first input type
-style the buttons

islam95 commented on Form-Controls/index.html:
```diff
@@ -1,28 +1,73 @@
+      <div>
+        <label for="lname">Last Name</label>
+        <input type="text" id="lname" name="lname" placeholder="Enter your last name" pattern="^[a-zA-Z].{1}.*" title="Name can only begin with alphabet and minimum of 2 letters" required>
+      </div>
```
It's up to you, but you could join these two into one control - `name` or Full name.

islam95 commented on Form-Controls/index.html:
```diff
@@ -1,28 +1,73 @@
+      </div>
+
+      <div id="btn-align">
+        <INPUT type="submit" value="Send" class="button"> <INPUT type="reset" class="button">
```
The `input` elements for the buttons are written in uppercase `INPUT`, which works in HTML, but it's more consistent and cleaner to use lowercase `<input>`.

islam95 commented on Form-Controls/index.html:
```diff
@@ -1,28 +1,73 @@
+        </select>
+      </div>
+      <div>
+        <label for="color">Size</label>
```
Can identify your mistake here? I think it's just a small mistake you made here because of a copy paste.

islam95 commented on Form-Controls/index.html:
```diff
@@ -1,28 +1,73 @@
+        <label for="color">Size</label>
+        <select id="size" name="size" required>
+          <option value="" disabled selected>Select a size</option>
+          <option value="XS">ES</option>
```
Another typo here :)

islam95 commented on Form-Controls/index.html:
```diff
@@ -1,28 +1,73 @@
+
+      <div>
+        <label for="email">Email Address</label>
+        <input type="email" id="email" placeholder="Enter your email address"  required>
```
something is missing in this input. Could you identify it?

islam95 commented on Form-Controls/styles.css:
```diff
@@ -0,0 +1,77 @@
+}
+
+input{
+  width: 400px;
```
a fixed width of 400px might be problematic on smaller screens. Consider using a percentage-based width or max-width to ensure responsiveness.

islam95 commented on Form-Controls/styles.css:
```diff
@@ -0,0 +1,77 @@
+  margin-left: 20px;
+}
+select{
+  width: 400px;
```
Similar to the input fields, avoid using `px` for width and `margin-left: 85px` to align the element. This hard-coded margin might break on different screen sizes.
You can set the width as 100% with max-width for responsiveness.

islam95 commented on Form-Controls/styles.css:
```diff
@@ -0,0 +1,77 @@
+  background: lightslategray; 
+  color: black;
+  cursor: pointer;
+  margin-left: 30%;
```
Instead of using `margin-left: 30%` for alignment, center the button using flexbox inside `#btn-align`

islam95 commented on Form-Controls/styles.css:
```diff
@@ -0,0 +1,77 @@
+  font-size: 20px;
+  margin-bottom: 10px;
+  padding: 2px;
+  border: 0px solid #222222;
```
The same here, 0px width border, means there is no border.

islam95 reviewed (changes requested):
Hi @HamzatKazeem Well done. It's your second PR!
You page can be looked at https://deploy-preview-216--cyf-ufd.netlify.app/form-controls/ 

Have a look at https://pagespeed.web.dev/analysis/ and check your page analysis. Try to address issues with Accessibility. You should aim to 100% on that one for this task.

HamzatKazeem committed:
Fixing PR Feedback

-fix width issues using max-width
-fixing typo errors
-fixing missing attributes within form elements

islam95 commented on Form-Controls/styles.css:
```diff
@@ -17,55 +17,56 @@ h1{
+  justify-content: center;
+}
+#btn-space{
+  margin-right: 8px;
```
instead of creating new `id` selector, you can add `gap: 8px` to  `#btn-align`.
You may read about gap property here - https://www.w3schools.com/cssref/css3_pr_gap.php

HamzatKazeem replied:
Thank you for this new knowledge.

islam95 reviewed (approved):
Well done @HamzatKazeem. You addressed my suggestions well.
You may now complete this PR.

HamzatKazeem commented:
@islam95 Thank you for your time and feedback

HamzatKazeem committed:
Remove Extra CSS Selector

Removing the extra CSS selector and replacing it with attribute gap to separate the buttons.
